python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -0,0 +1,370 @@
+# pylint: disable=anomalous-backslash-in-string
+import json
+from collections import defaultdict
+from typing import TYPE_CHECKING, Dict, List, Mapping, Optional, Sequence, Set, Tuple
+
+import toposort
+
+from dagster._annotations import experimental
+from dagster._core.storage.pipeline_run import IN_PROGRESS_RUN_STATUSES, RunsFilter
+
+from .asset_selection import AssetSelection
+from .events import AssetKey
+from .run_request import RunRequest
+from .sensor_definition import DefaultSensorStatus, MultiAssetSensorDefinition
+from .utils import check_valid_name
+
+if TYPE_CHECKING:
+    from dagster._core.definitions import AssetsDefinition, SourceAsset
+    from dagster._core.storage.event_log.base import EventLogRecord
+
+
+def _get_upstream_mapping(
+    selection,
+    assets,
+    source_assets,
+) -> Mapping[AssetKey, Set[AssetKey]]:
+    """Computes a mapping of assets in self._selection to their parents in the asset graph"""
+    upstream = defaultdict(set)
+    selection_resolved = list(selection.resolve([*assets, *source_assets]))
+    for a in selection_resolved:
+        a_parents = list(
+            AssetSelection.keys(a).upstream(depth=1).resolve([*assets, *source_assets])
+        )
+        # filter out a because upstream() includes the assets in the original AssetSelection
+        upstream[a] = {p for p in a_parents if p != a}
+    return upstream
+
+
+def _get_parent_updates(
+    context,
+    current_asset: AssetKey,
+    parent_assets: Set[AssetKey],
+    cursor_timestamp: float,
+    will_materialize_list: Sequence[AssetKey],
```

Comment:
Documenting the meaning of this argument would be helpful, because it took me a little while to figure it out.